Description: Running setup.py without LDFLAGS or explicitly passing includes
  will attempt to use bjam. This causes "python setup.py clean -a" to fail
  when building the package.
Origin: vendor
Author: Andrew Starr-Bochicchio <asb@debian.org>

--- libtorrent-rasterbar.orig/bindings/python/setup.py
+++ libtorrent-rasterbar/bindings/python/setup.py
@@ -51,7 +51,7 @@ except:
 ext = None
 packages = None
 
-if '--bjam' in sys.argv or ldflags == None or extra_cmd == None:
+if '--bjam' in sys.argv:
 
 	if '--bjam' in sys.argv:
 		del sys.argv[sys.argv.index('--bjam')]
@@ -97,15 +97,16 @@ else:
 	source_list = os.listdir(os.path.join(os.path.dirname(__file__), "src"))
 	source_list = [os.path.join("src", s) for s in source_list if s.endswith(".cpp")]
 
-	ext = [Extension('libtorrent',
-		sources = source_list,
-		language='c++',
-		include_dirs = parse_cmd(extra_cmd, '-I'),
-		library_dirs = parse_cmd(extra_cmd, '-L'),
-		extra_link_args = ldflags.split() + arch(),
-		extra_compile_args = parse_cmd(extra_cmd, '-D', True) + arch() \
-			+ target_specific(),
-		libraries = ['torrent-rasterbar'] + parse_cmd(extra_cmd, '-l'))]
+	if extra_cmd:
+		ext = [Extension('libtorrent',
+			sources = source_list,
+			language='c++',
+			include_dirs = parse_cmd(extra_cmd, '-I'),
+			library_dirs = parse_cmd(extra_cmd, '-L'),
+			extra_link_args = ldflags.split() + arch(),
+			extra_compile_args = parse_cmd(extra_cmd, '-D', True) + arch() \
+				+ target_specific(),
+			libraries = ['torrent-rasterbar'] + parse_cmd(extra_cmd, '-l'))]
 
 setup(name = 'python-libtorrent',
 	version = '1.0.9',
